addons/mail/models/mail_thread.py

```diff
@@ -2536,7 +2536,16 @@ def message_post_with_source(self, source_ref, render_values=None,
         messages_all = self.env['mail.message']
         for record in self:
             if template:
+                # Notify author if email_cc points to him
+                if self.env.context.get('mail_notify_author'):
+                    notify_author = True
+                elif self.env.user.company_id.email:
+                    notify_author = self.env.user.company_id.email in template._render_field('email_cc', record.ids)[record.id]
```

Comment:
it would make more sense to me to have `composer.author_id.email`, as far as I can tell the author_id can be different than the current user company_id (eg. if you send the mail from backend, the author_id is the current user), so if we send a mail.template from backend (or any other partner than the company) we would add `mail_notify_author` which will send the mail to the author too (that may be different than company_id.partner_id)

small side note: since we do `in` we would also match "john@doe.com" if in the template we had "mr.john@doe.com" which is not much likely but would be annoying (but fixing that would be annoying, since we don't have a list of "partner_cc" but partner that we don't know if we need to CC or not)

Reply:
Thanks for inspecting it again. I'm going to try using `composer.author_id.email`

I went with `in` cause render will give email address with a display name (eg. `"John Doe <john.doe@example.com>"`)
I see that https://github.com/odoo/odoo/blob/d924341c6fc9556339042769f2c6516ea4b1031d/odoo/tools/mail.py#L541 looks like something that can make it more robust